### PR TITLE
Replace metaSnapshot with suffixSnapshot

### PR DESCRIPTION
**Pull request by congqixia**
Replace metaSnapshot with suffixSnapshot

Related issue: #7530 #7945

Signed-off-by: Congqi Xia <congqi.xia@zilliz.com>

### CODE REVIEW

File: internal/rootcoord/root_coord.go [hunk 1]
```diff
@@ -855,24 +858,34 @@ func (c *Core) Register() error {
...
+	}
 	c.initOnce.Do(func() {
 		connectEtcdFn := func() error {
 			if c.etcdCli, initError = clientv3.New(clientv3.Config{Endpoints: Params.EtcdEndpoints, DialTimeout: 5 * time.Second}); initError != nil {
 				log.Error("RootCoord, Failed to new Etcd client", zap.Any("reason", initError))
 				return initError
 			}
-			var ms *metaSnapshot
-			ms, initError = newMetaSnapshot(c.etcdCli, Params.MetaRootPath, TimestampPrefix, 1024)
+			if c.kvBase, initError = c.kvBaseCreate(Params.KvRootPath); initError != nil {
```
Comment (xiaofan-luan) [inline internal/rootcoord/root_coord.go:872]: what is the reason why we need. kvBaseCreate function? 
why not just etcdkv.NewEtcdKV?

File: internal/rootcoord/root_coord.go [hunk 2]
```diff
@@ -855,24 +858,34 @@ func (c *Core) Register() error {
...
+				log.Error("RootCoord, Failed to new EtcdKV", zap.Any("reason", initError))
+				return initError
+			}
+			var metaKV kv.TxnKV
+			metaKV, initError = c.kvBaseCreate(Params.MetaRootPath)
```
Comment (xiaofan-luan) [inline internal/rootcoord/root_coord.go:877]: Don't think we should start multiple etcd client here, it shoud be a singleton.
Metakv and snapshotkv just has different root paths.

File: internal/rootcoord/root_coord.go (hunk 1 above)
Comment (congqixia) [reply internal/rootcoord/root_coord.go:872]: `kvBaseCreate` can provide injection point for changing `TxnKV` implementation, which provided a method to inject failure in ut.

File: internal/rootcoord/root_coord.go (hunk 2 above)
Comment (congqixia) [reply internal/rootcoord/root_coord.go:877]: Agree. But kv package does not have a way to create a `EtcdKV` with existing etcdclient. Will have a PR fixing this,

**Pull request merged**